### ISSUE TITLE
Strange SOA Record on deSEC DNS

### ISSUE THREAD

**Issue by logopk** [desec-io/desec-stack#259]
First of all, is this the right spot to raise an operational issue? Please advise if not.

When I test my Domain using viewdns.info (or mxtoolbox.com) I get warning messages on the SOA record:

Oops! The primary nameserver listed in your SOA record (**set.an.example.**) is not listed at the parent servers! This could suggest a configuration issue with your SOA record.
--

Oops! Your SOA serial number (201910**56**03) doesn't seem to be in the recommended format (YYYYMM**DD**nn - where nn is the revision number). This is still OK, however as long as you are keeping track of your SOA version details.
--

https://viewdns.info/dnsreport/?domain=dedyn.io (I put dedyn.io as an example, my domain has a similar SIOA)

Apparently this is not set by myself:
> SOA record
The SOA record cannot be read or written through this interface. When attempting to create, 
modify or otherwise access an SOA record, 400 Bad Request or 403 Forbidden is returned, respectively.
The rationale behind this is that the content of the SOA record is entirely determined by the DNS operator, and users should not have to bother with this kind of metadata. Upon zone changes, the backend automatically takes care of updating the SOA record accordingly.
If you are interested in the value of the SOA record, you can retrieve it using a standard DNS query.

Any reason or solution for this?

Thanks

Peter

**Comment by peterthomassen** [comment]
Hi Peter, thanks for the report! Yes, this is the right spot to raise issues like that.

tl;dr: Both findings are non-issues.

1.) The hostname in the SOA can be used for slave name servers who have a copy of the DNS zone in order to figure out which nameserver to ask for updates. It's supposed to be the server with the "master copy". However, as we organize our master-slave synchronization differently (and regular clients/resolvers never interpret this value), this hostname is meaningless.

For details, check out [RFC 1035 Sec. 3.3.13](https://tools.ietf.org/html/rfc1035#section-3.3.13). It defines that this hostname is the "<domain-name> of the name server that was the original or primary source of data for this zone." Note that this does not require the domain name to be public; it could be an internal hostname within the DNS operator's network, for example. It bears no public meaning.

2.) Regarding the serial: The only important thing is that is increases when a DNS record changes, see the same RFC section as cited above: The serial is the "unsigned 32 bit version number of the original copy of the zone.  Zone transfers preserve this value.  This value wraps and should be compared using sequence space arithmetic." We maintain this requirement.

Many DNS operators use serials in the YYYYMMDDnn format, but there's no requirement by this (neither by specification, nor practically). Our serials are the sum of the YYYYMMDDnn-type serial and the number of weeks that has passed since Jan 1st, 1970. This means that our serials increase weekly. The purpose of this is to cause a serial increment whenever our DNSSEC signatures are updated (which we do weekly). For details, see https://doc.powerdns.com/authoritative/dnssec/operational.html#possible-soa-edit-values

So, to sum up, both findings are related to somewhat unconventional decisions we made, but they have no operational downsides, and they are RFC-compliant. They are thus false positives, and the tool you used is overly sensitive. :-)

**Comment by logopk** [comment]
Perfect, Thanks. Nevertheless the "set.an.example" is not really obvious. Any good place to document this?

**Comment by peterthomassen** [comment]
If the issue recurs, we'll add some info to the docs explaining that our SOA records are compliant (even though some tools raise some unjustified objections). :)
